tests/ops/service/connectors/saas/test_authenticated_client.py

```diff
@@ -145,6 +176,19 @@ def test_client_ignores_errors(
             errors_to_ignore=[401],
         )
 
+    def test_sending_special_characters(
+        self, test_authenticated_client, test_http_server
+    ):
+        request_params = SaaSRequestParams(
+            method=HTTPMethod.POST,
+            path="/",
+            body='{"addr": "1234 Peterson’s Farm Rd."}',
+            headers={"Content-Type": "application/json"},
+        )
+
+        test_authenticated_client.uri = test_http_server
+        test_authenticated_client.send(request_params)
```

Comment:
I'm spinning up a minimal test server so that the `send()` can get past the connection errors and actually get to the part of its code where it attempts to parse the body. I wasn't able to reproduce the error when using mocks.